sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/autoconfigure/storage/resource/AzureStorageBlobProtocolResolver.java

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.autoconfigure.storage.resource;
+
+import com.azure.storage.blob.BlobServiceClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ResourceLoaderAware;
+import org.springframework.core.io.DefaultResourceLoader;
+import org.springframework.core.io.ProtocolResolver;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+
+/**
+ * A {@link ProtocolResolver} implementation for the {@code azure-blob://} protocol.
+ *
+ */
+public class AzureStorageBlobProtocolResolver implements ProtocolResolver, ResourceLoaderAware {
```

Comment:
I don't think this should be here, probabaly a much lower level module

core? add storage as an optional dependency?
what do you think?

Reply:
What we need here is something like spring-cloud-azure-context, this shouldn't be included in core.